### PR TITLE
Teams: Add organization menu item for user dashboard

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@codenvy.com>

### What does this PR do?
Add organizations menu item to navbar.

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/1858
https://github.com/codenvy/codenvy/issues/1859
https://github.com/codenvy/codenvy/issues/1860
https://github.com/codenvy/codenvy/issues/1912
https://github.com/codenvy/codenvy/issues/1922
https://github.com/codenvy/codenvy/issues/1930
https://github.com/codenvy/codenvy/issues/1614

#### Changelog
Add organization menu item for system admins only.

#### Release Notes
Add organization menu item for system admins only. This allows us to manage organizations(create, delete, edit, create a sub one and manage their resources. Organizations allow groups of developers to collaborate with private and shared workspaces. Resources and permissions are controlled and allocated within the organization by administrator.

#### Docs PR
[in progress]

selection_032
selection_031
selection_030
selection_028
selection_024
selection_022

### CODE REVIEW

**Comment by ashumilova** [comment]
Feedbacks : 
- [ ] on organization/sub-organization creation redirect to its details page (not list) as we do with other objects
- [ ] validate the name of the organization (exists or not?) on creation and edit
- [ ] on save organization details - do not hide all inputs - just display loader

File: dashboard/src/app/organizations/organization-details/organization-members/list-organization-members.controller.ts [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+   * @param member
+   *
+   * @returns {boolean}
+   */
+  isOwner(member: any): boolean {
+    if (!this.organization || !member) {
+      return false;
+    }
+
+    return this.organization.qualifiedName.split('/')[0] === member.userId;
```
Comment (ashumilova) [inline dashboard/src/app/organizations/organization-details/organization-members/list-organization-members.controller.ts]: Need to compare with user name

File: dashboard/src/components/api/codenvy-organization-events-manager.factory.ts [hunk 2]
```diff
@@ -157,22 +157,22 @@ export class CodenvyTeamEventsManager {
...
    */
-  processRenameTeam(info: any): void {
+  processRenameOrganization(info: any): void {
     let isCurrentUser = this.isCurrentUser(info.initiator);
     if (isCurrentUser) {
       //TODO
     } else {
-      let title = 'Team renamed';
-      let content = 'Team \"' + info.oldName + '\" has been renamed to \"' + info.newName + '\" by ' + info.initiator;
+      let title = 'Organization/Team renamed';
```
Comment (ashumilova) [inline dashboard/src/components/api/codenvy-organization-events-manager.factory.ts]: -1 on message format "**Organization/Team** test has been renamed ..."
@slemeur do we need live notifications for organizations in onprem?

File: dashboard/src/app/organizations/create-organizations/create-organizations.controller.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  /**
+   * Loading state of the page.
+   */
+  private isLoading: boolean;
+  /**
+   * The list of users to invite.
+   */
+  private members: Array<any>;
+  /**
+   * Owner'e email.
```
Comment (olexii4) [inline dashboard/src/app/organizations/create-organizations/create-organizations.controller.ts]: done

File: dashboard/src/app/organizations/create-organizations/create-organizations.controller.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+   */
+  private owner: string;
```
Comment (olexii4) [inline dashboard/src/app/organizations/create-organizations/create-organizations.controller.ts]: done

File: dashboard/src/app/organizations/create-organizations/create-organizations.controller.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        };
+
+        let promise = this.codenvyPermissions.storePermissions(permissions);
+        promises.push(promise);
+      }
+    });
+
+    this.$q.all(promises).then(() => {
+      this.isLoading = false;
+      let path = '/organization/';
```
Comment (olexii4) [inline dashboard/src/app/organizations/create-organizations/create-organizations.controller.ts]: done

File: dashboard/src/app/organizations/organization-details/organization-members/list-organization-members.controller.ts (hunk 1 above)
Comment (olexii4) [inline dashboard/src/app/organizations/organization-details/organization-members/list-organization-members.controller.ts]: done

File: dashboard/src/components/api/codenvy-organization-events-manager.factory.ts (hunk 2 above)
Comment (olexii4) [inline dashboard/src/components/api/codenvy-organization-events-manager.factory.ts]: done

File: dashboard/src/app/teams/team-details/team-details.controller.ts [hunk 6]
```diff
@@ -112,12 +112,12 @@ export class TeamDetailsController {
    */
   constructor(codenvyTeam: CodenvyTeam, codenvyResourcesDistribution: CodenvyResourcesDistribution, codenvyPermissions: CodenvyPermissions,
               codenvyUser: CodenvyUser, $route: ng.route.IRouteService, $location: ng.ILocationService, $rootScope: che.IRootScopeService,
-              $scope: ng.IScope, confirmDialogService: any, codenvyTeamEventsManager: CodenvyTeamEventsManager, cheNotification: any,
+              $scope: ng.IScope, confirmDialogService: any, codenvyOrganizationEventsManager: CodenvyTeamEventsManager, cheNotification: any,
```
Comment (ashumilova) [inline dashboard/src/app/teams/team-details/team-details.controller.ts]: will not work,need to change : codenvyOrganizationEventsManager -> codenvyTeamEventsManager

Comment (olexii4) [inline dashboard/src/app/teams/team-details/team-details.controller.ts]: Typo. 
Done

**Comment by slemeur** [comment]
Here is the list of issues identified and discussed during our review session call.

## Issues:
- [ ] Need to display the resources and caps for root-organizations
- [ ] It must not be possible to add members in a sub-organisation if they are not members of the organization
- [ ] As a user, when navigating to Organization menu, I should not be able to add organization or sub-organization
- [ ] As a user, when navigating to Organization menu, I should be able to know how much resources each organizations are providing (need to add .
- [ ] As a user, I should be able to know how much resources I can use (for personal workspaces and organization) @ashumilova is checking that.

## Email management:
- [ ] All email notification templates must be adapted for organization (instead of teams)

## User management:
- [ ] As a system admin, I must be able to see the list of organization/sub-org for a particular user (need a user's details screen)
- [ ] As a system admin, I must be able to quickly add a user to an organization/sub-org from the user's details screen


## Typo and labels changes
- [ ] Replace "Developers" by "Members" in organization and sub-organization
screen_shot_2017-03-27_at_14_25_36

screen_shot_2017-03-27_at_14_17_45

- [ ] Update labels on screen to create a new organization
screen_shot_2017-03-27_at_14_22_58

- [ ] Update labels on screen to create a new sub-organization
screen_shot_2017-03-27_at_14_20_29

- [ ] Update description of "Administrator" role for an organization
screen_shot_2017-03-27_at_14_16_44

- [ ] Update labels when browsing list of sub-organizations
screen_shot_2017-03-27_at_14_15_25

## UI and UX issues

- [ ] Loading of sub-organization details is flickering
previous-content-displayed

- [ ] Add navigation from sub-organization details screen. 
screen_shot_2017-03-27_at_12_24_02
(Keep the title as it for now.)

- [ ] In "Organization" for user, we need to display the list of all organizations/sub-organization as a plain list.

**Comment by akurinnoy** [comment]
The list of organization should be refreshed after an organization removal.

anim

**Comment by olexii4** [comment]
- [ ] We shouldn't have organization menu for non admin user in SAAS.
selection_022

**Comment by olexii4** [comment]
- [ ] We 
selection_020
shouldn't have root team and personal one

**Comment by akurinnoy** [comment]
- [ ] Role should be changed only if request is successful

anim

- [ ] parent organization admin should be able to manage resources of sub-organization where he is a member

anim2

- [ ] Number of organizations in sidebar should be equal to number of organizations in the list

screenshot-a1 codenvy-stg com-2017-04-11-14-14-29

**Comment by Ohrimenko1988** [comment]
Need to add some view for managing using resources ,  in current time one unscrupulous organization member able to block organization by using all organization resources .

**Comment by olexii4** [comment]
We have 'back' button 
selection_003
 with wrong URL in case when sub-organization name looks like to some part of its parent.
For example:
Sub-organization name is 'test' in case with parent name 'test-sub'
./organization/test-root/test-sub/test.

**Comment by olexii4** [comment]
Sorry. Accidentally closed and then reopened.

**Comment by SkorikSergey** [comment]
Problems with the Bulk checkbox while add a member to an organization.
anim

**Comment by SkorikSergey** [comment]
User can still work with created workspace after deleted from an organization.
anim

**Comment by olexii4** [comment]
selection_006

**Comment by ashumilova** [comment]
image

**Comment by ashumilova** [comment]
Added available RAM info on workspace creation:
image

**Pull request merged**